lib/dams/controller_helper.rb

```diff
@@ -675,8 +675,12 @@ def audit( id = "unknown" )
     def referrer_controller( request )
       uri = URI(request.referrer || "")
       if uri.host == request.host
```

Comment:
Yes, it sounds good. We can try to see whether other ways will work since Rails.application.routes.recognize_path() do throw the routing exception and the begin rescue block does seem to be correct here. But if we think about referrer_controller method here is just try to parse the referrer URL and return the controller if there is a matched otherwise return nil sounds easier to understand.